contracts/consensus-gateway/ConsensusCogateway.sol

```diff
@@ -85,4 +92,141 @@ contract ConsensusCogateway is MasterCopyNonUpgradable, MessageBus, ConsensusGat
             _maxStorageRootItems
         );
     }
+
+    /**
+     * @notice Confirm deposit in order to mint tokens.
+     *
+     * @param _amount MOST token deposit amount in atto.
+     * @param _beneficiary Address of beneficiary on auxiliary chain.
+     * @param _feeGasPrice Fee gas price at which rewards will be calculated.
+     * @param _feeGasLimit Fee gas limit at which rewards will be calculated.
+     * @param _depositor Address of depositor on the origin chain.
+     * @param _blockHeight Block height of origin chain at which storage proof
+     *                     is generated.
+     * @param _rlpParentNodes Storage merkle proof to verify message declaration
+     *                        on the origin chain.
+     *
+     * @return messageHash_ Message hash.
+     */
+    function confirmDeposit(
+        uint256 _amount,
+        address payable _beneficiary,
+        uint256 _feeGasPrice,
+        uint256 _feeGasLimit,
+        address _depositor,
+        uint256 _blockHeight,
+        bytes calldata _rlpParentNodes
+    )
+        external
+        returns (bytes32 messageHash_)
+    {
+
+        require(
+            _amount != 0,
+            "Deposit amount should be greater than 0"
+        );
+        require(
+            _beneficiary != address(0),
+            "Beneficiary address must not be 0."
+        );
+        require(
+            _depositor != address(0),
+            "Depositor address must not be 0."
+        );
+
+        uint256 initialGas = gasleft();
+
+        bytes32 depositIntentHash = ERC20GatewayBase.hashDepositIntent(
+            _amount,
+            _beneficiary
+        );
+
+        messageHash_ = confirmMessage(
+            _depositor,
+            depositIntentHash,
+            _feeGasPrice,
+            _feeGasLimit,
+            _blockHeight,
+            _rlpParentNodes
+        );
+
+        uint256 gasConsumed = initialGas.sub(gasleft());
```

Comment:
there can be a constant added to accomodate for the remaining executions; alternatively this can be absorbed in the gas price a facilitator accepts, but it is less clear imo

Reply:
I added a comment to explain that additional gas consumption can be adjusted with gas price value.

Reply:
ok

@jayesh can you please note a task to correct the gas calculation in Generation2, when addressing reward structures. 🙏